crates/repl/src/kernels.rs

```diff
@@ -465,6 +467,123 @@ async fn read_kernels_dir(path: PathBuf, fs: &dyn Fs) -> Result<Vec<LocalKernelS
     Ok(valid_kernelspecs)
 }
 
+#[allow(dead_code)]
+pub fn python_env_kernel_specifications(
+    project: &Model<Project>,
+    worktree_id: WorktreeId,
+    cx: &mut AppContext,
+) -> impl Future<Output = Result<Vec<KernelSpecification>>> {
+    let fs = project.read(cx).fs().clone();
+    let python_language = LanguageName::new("Python");
+
+    let toolchains = project
+        .read(cx)
+        .available_toolchains(worktree_id, python_language, cx);
+
+    async move {
```

Comment:
These should be setup as futures that can await in parallel rather than serially.